tests/test_smoke.py

```diff
@@ -1189,9 +1190,17 @@ def test_docker_storage_mounts(generic_cloud: str, image_id: str):
     template = jinja2.Template(template_str)
     # ubuntu 18.04 does not support fuse3, and blobfuse2 depends on fuse3.
     azure_mount_unsupported_ubuntu_version = '18.04'
+    # Commands to verify bucket upload. We need to check all three
+    # storage types because the optimizer may pick any of them.
+    s3_command = f'aws s3 ls {storage_name}/hello.txt'
+    gsutil_command = f'gsutil ls gs://{storage_name}/hello.txt'
+    azure_blob_command = TestStorageWithCredentials.cli_ls_cmd(
+        storage_lib.StoreType.AZURE, storage_name, suffix='hello.txt')
     if azure_mount_unsupported_ubuntu_version in image_id:
+        include_private_mount = False if generic_cloud == 'azure' else True
```

Comment:
can we add a comment for why private bucket is not allowed for azure?

Reply:
Added!

Reply:
Let's file an issue for this to check the dependencies earlier in the `setup.py`?